src/Microsoft.Data.Analysis/TextFieldParser.cs

```diff
@@ -0,0 +1,642 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Data.Analysis
+{
+    public enum FieldType
+    {
+        Delimited,
+        FixedWidth
+    }
+
+    public class TextFieldParser : IDisposable
```

Comment:
We could in theory split this off as a separate csproj to make its own Nuget package. Something to consider ...